src/mqtt/mod.rs

```diff
@@ -218,6 +225,109 @@ async fn listen_on_camera(cam_config: Arc<CameraConfig>, mqtt_config: &MqttConfi
     Ok(())
 }
 
+/// Enables MQTT discovery for a camera. See docs at https://www.home-assistant.io/integrations/mqtt/#mqtt-discovery
+async fn enable_discovery(
+    discovery_config: &MqttDiscoveryConfig,
+    mqtt_sender: &MqttSender,
+    cam_config: &Arc<CameraConfig>,
+) -> Result<()> {
+    debug!("Enabling MQTT discovery for {}", cam_config.name);
+
+    let mut connections = array![];
+    if let Some(addr) = &cam_config.camera_addr {
+        connections
+            .push(array!["camera_addr", addr.clone()])
```

Comment:
Can we do this as an enum. 

```rust
enum CamAddr {
  Addr(SocketAddr),
  Uid(String)
}

// later in the fn

connections.push(CamAddr::Addr(addr.parse()?);

// and the next one

connections.push(CamAddr::Uid(uid.clone));
```

Reply:
You should also handle the case where it is a host name and it resolves to multiple addresses.

Reply:
Hmm... not following I think.
- `cam_config.camera_addr` is actually a `String` 
- `connections` is just a `JsonValue` that we end up dumping on the discovery topic to uniquely identify a device. So I think we can just take the strings as-is and pass them rather than parsing/resolving them? That would allow the MQTT config to be easily mapped back to the Neolink config.

Reply:
Ok I got you as it is part of the discovery spec

Reply:
Yep that's right, sorry that should have been more obvious!